lib/rouge/lexers/graphql.rb

```diff
@@ -20,19 +20,25 @@ class GraphQL < RegexLexer
         end
 
         rule %r/\bfragment\b/, Keyword, :fragment_definition
+        rule %r/\bscalar\b/, Keyword, :value
+        rule %r/\bextend\sscalar\b/, Keyword, :value
 
         rule %r/\b(?:type|interface|enum)\b/, Keyword, :type_definition
+        rule %r/\b(?:extend\s+type|extend\s+interface|extend\s+enum)\b/, Keyword, :type_definition
+
         rule %r/\b(?:input|schema)\b/, Keyword, :type_definition
+        rule %r/\b(?:extend\sinput|extend\sschema)\b/, Keyword, :type_definition
 
         rule %r/\bunion\b/, Keyword, :union_definition
+        rule %r/\bextend\s+union\b/, Keyword, :union_definition
 
         mixin :basic
 
         # Markdown descriptions
         rule %r/(""")(.*?)(""")/m do |m|
-          token Str::Double, m[1]
-          delegate Markdown, m[2]
-          token Str::Double, m[3]
+          token Str::Double, m[1] + "\n"
```

Comment:
I don't think this is a good way to achieve this. The lexer shouldn't need to rewrite the values that are matched by rule. If you want to keep the starting and ending newlines out of the value passed to the Markdown lexer, it would be better to change the regular expression.

So you'd have something like:

```ruby
rule %r/(""")(\n)(.*?)(\n)(""")/m do |m|
  token Str::Double, m[1]
  token Text::Whitespace, m[2]
  delegate Markdown, m[3]
  token Text::Whitespace, m[4]
  token Str::Double, m[5]
end
```